### PR TITLE
Added `prompt` only `getCompletions` convenience methods

### LINKED ISSUE #34671: [OpenAI] Add an overload API with "Prompt" var as required for getCompletions endpoint
See JS for example,
https://github.com/trrwilson/azure-sdk-for-net/blob/user/travisw/aoai-chat-api/sdk/openai/Azure.AI.OpenAI/tests/SessionRecords/OpenAIInferenceTests/Completions(Azure).json

### PR DESCRIPTION
**Pull request by jpalvarezl**
# Description
Closes #34671

This PR Adds the following:
- Convenience method requires only `List<String> prompts` for `getCompletions`
- Unit tests
- Samples

### CODE REVIEW

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java [hunk 1]
```diff
@@ -180,6 +183,94 @@ public Mono<Response<BinaryData>> getCompletionsWithResponse(
...
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return completions for the provided input prompts. Completions support a wide variety of tasks and generate text
+     *     that continues from or "completes" provided prompt data along with {@link Response} on successful completion
+     *     of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<BinaryData>> getCompletionsWithResponse(
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:267]: We already has a max-overload method getCompletionWithResponse(),
Adding getCompletions(String deploymentId, String prompt) should be enough? what do you think?

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/CompletionsUtils.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+     * Convenience method for minimal initialization for the CompletionsOptions class
+     * @param prompt from which Completions will be generated
+     * @return A CompletionsOptions object */
+    public static CompletionsOptions DefaultCompletionsOptions(String prompt) {
+        List<String> prompts = new ArrayList<>();
+        prompts.add(prompt);
+        return new CompletionsOptions(prompts).setMaxTokens(DEFAULT_MAX_COMPLETION_TOKENS);
+    }
+
+    public static int DEFAULT_MAX_COMPLETION_TOKENS = 100;
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/CompletionsUtils.java:23]: public static final

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java [hunk 3]
```diff
@@ -417,9 +417,9 @@ public Mono<Completions> getCompletions(String deploymentId, CompletionsOptions
      *     that continues from or "completes" provided prompt data on successful completion of {@link Mono}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Completions> getCompletions(String deploymentId, List<String> prompts) {
+    public Mono<Completions> getCompletions(String deploymentId, String prompt) {
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:420]: We don't need to add the `getCompletionsWithResponse`  method above that returns the BinaryData for a single prompt.

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java [hunk 4]
```diff
@@ -406,7 +406,7 @@ public Mono<Completions> getCompletions(String deploymentId, CompletionsOptions
      * that continues from or "completes" provided prompt data.
      *
      * @param deploymentId deployment id of the deployed model.
-     * @param prompts The prompts to generate values from.
+     * @param prompt The prompts to generate values from.
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:409]:
```suggestion
     * @param prompt The prompt to generate values from.
```

Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:409]: Update the docs on line 405 as well to make `prompts` singular.

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java [hunk 5]
```diff
@@ -249,7 +249,7 @@ public Response<BinaryData> getCompletionsWithResponse(
      * }</pre>
      *
      * @param deploymentId deployment id of the deployed model.
-     * @param prompts The prompts to generate values from.
+     * @param prompt The prompts to generate values from.
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java:252]:
```suggestion
     * @param prompt The prompt to generate values from.
```

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java [hunk 6]
```diff
@@ -260,8 +260,8 @@ public Response<BinaryData> getCompletionsWithResponse(
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<BinaryData> getCompletionsWithResponse(
-        String deploymentId, List<String> prompts, RequestOptions requestOptions) {
-        CompletionsOptions completionsOptions = CompletionsUtils.DefaultCompletionsOptions(prompts);
+        String deploymentId, String prompt, RequestOptions requestOptions) {
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java:263]: This overload is not needed.

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java [hunk 7]
```diff
@@ -310,6 +401,30 @@ public Mono<Completions> getCompletions(String deploymentId, CompletionsOptions
...
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return completions for the provided input prompts. Completions support a wide variety of tasks and generate text
+     *     that continues from or "completes" provided prompt data on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Completions> getCompletions(String deploymentId, String prompt) {
+        RequestOptions requestOptions = new RequestOptions();
+        CompletionsOptions completionsOptions = CompletionsUtils.DefaultCompletionsOptions(prompt);
+        return getCompletionsWithResponse(deploymentId, BinaryData.fromObject(completionsOptions), requestOptions)
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:423]: Call the convenience method instead `getCompletions(deploymentId, completionOptions)`, so you don't have to repeat the code to convert BinaryData into Completions.

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/CompletionsUtils.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+// Licensed under the MIT License.
+
+package com.azure.ai.openai.implementation;
+
+import com.azure.ai.openai.models.CompletionsOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class CompletionsUtils {
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/CompletionsUtils.java:11]: Add class-level javadoc string.

File: sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/GetCompletionsFromPrompt.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+import com.azure.ai.openai.models.Choice;
+import com.azure.ai.openai.models.Completions;
+import com.azure.core.credential.AzureKeyCredential;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sample demonstrates the minimal use we can have of the SDK where the user, aside from providing authentication
+ * details, only needs to pass the list of prompts and the deploymentId
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/GetCompletionsFromPrompt.java:15]: it mentiones "list of prompts" but should be one single prompt in real.

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java [hunk 10]
```diff
@@ -313,6 +315,27 @@ public Mono<Completions> getCompletions(String deploymentId, CompletionsOptions
                 .map(protocolMethodData -> protocolMethodData.toObject(Completions.class));
     }
 
+    /**
+     * Gets completions for the provided input prompt. Completions support a wide variety of tasks and generate text
+     * that continues from or "completes" provided prompt data.
+     *
+     * @param deploymentId deployment id of the deployed model.
+     * @param prompt The prompt to generate completion text from..
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIAsyncClient.java:323]:
```suggestion
     * @param prompt The prompt to generate completion text from.
```

File: sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/CompletionsUtils.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+    public static final int DEFAULT_MAX_COMPLETION_TOKENS = 100;
+
+    private CompletionsUtils() {
+    }
+
+    /**
+     * Convenience method for minimal initialization for the CompletionsOptions class
+     * @param prompt from which Completions will be generated
+     * @return A CompletionsOptions object */
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/implementation/CompletionsUtils.java:21]:
```suggestion
     * @return A CompletionsOptions object
     */
```

File: sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/GetCompletionsFromPromptAsync.java [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+                completions -> {
+                    for (Choice choice : completions.getChoices()) {
+                        System.out.printf("%s.%n", choice.getText());
+                    }
+                },
+                error -> System.err.println("There was an error getting completions." + error),
+                () -> completionSink.emitEmpty(Sinks.EmitFailureHandler.FAIL_FAST)
+            );
+
+        completionSink.asMono().block();
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/GetCompletionsFromPromptAsync.java:51]: For an async example, we want to show customers how to use it asynchronously without blocking. Adding Mono `.timeout()` and `.block()` will make noise and confusion.

[the remark this answers is not in the document]
Comment (jpalvarezl) [reply sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/GetCompletionsFromPromptAsync.java:51]: Gotcha! My point was that `sleep()` is just as blocking as calling `block()` but I totally see your point of having confusing language in the sample. I will adjust it accordingly.

**Review by mssfang — APPROVED** [review]

**Pull request merged**